### DISCUSSION TITLE
Timeout Usage

### DISCUSSION THREAD

**Discussion by sarogersz** [khoih-prog/AsyncHTTPRequest_Generic#28] [Q&A]
Clearly there is a timeout capability in AsyncHTTPRequest_Generic, which is set via the method setTimeout() on a request basis.  There is also an enum with error codes that include:
#define HTTPCODE_TIMEOUT             (-11)
so it appears that a timeout is handled as an error.

I am at a point now where I'd like to incorporate error handling functionality in my HTTP code.  Thus far, I am unable to locate any examples or explanation of how to do this.

If there is anyone who knows how to do this, I'd be appreciative if you could share an example.

**Comment by SeeliSoft-CH** [comment]
When creating the request, you provide the ready state callback function:
`request.onReadyStateChange(requestCB);`

in this callback method, you only need to handle the "done" state. in case of a timeout, the http code will have value -11.

```
void requestCB(void* optParm, AsyncHTTPRequest* request, int readyState) {
    if (readyState == readyStateDone) {
        if (request->responseHTTPcode() < 0) {
            // Something wrong happened
        } else {
            // Request was technically ok, but it could still have returned e.g. error 400 or 500
    }
}
```
